src/decoder/flac.rs

```diff
@@ -145,27 +611,130 @@ where
                 return Some(real_val);
             }
 
-            // Load the next block.
+            // Cold path: need to decode next block
             self.current_block_off = 0;
             let buffer = mem::take(&mut self.current_block);
-            match self.reader.blocks().read_next_or_eof(buffer) {
+            match self
+                .reader
+                .as_mut()
+                .expect("reader should always be Some")
+                .blocks()
+                .read_next_or_eof(buffer)
+            {
                 Ok(Some(block)) => {
                     self.current_block_channel_len = (block.len() / block.channels()) as usize;
                     self.current_block = block.into_buffer();
                 }
-                _ => return None,
+                Ok(None) | Err(_) => {
+                    // No more blocks or error, current_block becomes empty
+                    // (buffer was consumed by read_next_or_eof)
+                    return None;
+                }
             }
         }
     }
+
+    /// Returns bounds on the remaining length of the iterator.
+    ///
+    /// Provides accurate size estimates based on FLAC metadata when available.
+    /// This information can be used by consumers for buffer pre-allocation
+    /// and progress indication.
+    ///
+    /// # Returns
+    ///
+    /// A tuple `(lower_bound, upper_bound)` where:
```

Comment:
The meaning of the tuple is specified by the Iterator trait. Unless we are deviating from that users should depend on the Iterator docs not a doc comment in the implementation. If we are deviating from the Iterator trait that we should add our own method.